phoenix-core/src/main/antlr3/PhoenixSQL.g

```diff
@@ -606,8 +606,8 @@ drop_index_node returns [DropIndexStatement ret]
 // Parse a alter index statement
 alter_index_node returns [AlterIndexStatement ret]
     : ALTER INDEX (IF ex=EXISTS)? i=index_name ON t=from_table_name
-      ((s=(USABLE | UNUSABLE | REBUILD | DISABLE | ACTIVE)) (async=ASYNC)? ((SET?)p=fam_properties)?)
-      {ret = factory.alterIndex(factory.namedTable(null, TableName.create(t.getSchemaName(), i.getName())), t.getTableName(), ex!=null, PIndexState.valueOf(SchemaUtil.normalizeIdentifier(s.getText())), async!=null, p); }
+      ((s=(USABLE | UNUSABLE | REBUILD | DISABLE | ACTIVE)) (all=ALL)? (async=ASYNC)? ((SET?)p=fam_properties)?)
```

Comment:
Grammar seems misleading. We are expecting ALL only after REBUILD keyword. Shouldn't it be a part of REBUILD or condition, as we don't support ALL on any other keyword like USABLE | DISABLE etc?

Reply:
ALL is similar to ASYNC and follows the same pattern.

Reply:
I am not sure if they should be handled in the same way? @karanmehta93 thoughts?